yang/frr-isisd.yang

```diff
@@ -2078,6 +2109,45 @@ module frr-isisd {
         }
       }
 
+      container route_leanking {
```

Comment:
We don't need this new container to model IS-IS to IS-IS redistribution. It is enough to modify the existing `redistribute` container:
- remove `must ". != \"isis\"";` from `leaf protocol` to allow redistribution from IS-IS
- add a new `from-level` leaf to configure from which level to redistribute:
  ```
  leaf from-level {
    type level;
    when "../protocol = \"isis\"";
    mandatory true;
    must "(. != \"level-1-2\") and ((../../../is-type = \"level-1-2\") and (. != ../level))";
    description
      "IS-IS level from which the routes should be redistributed.";
  }
  ```

These two changes would be enough to model what you need without introducing a lot of new NB callbacks.